pkg/controllers/machinesetsync/machineset_sync_controller_test.go

```diff
@@ -279,6 +309,49 @@ var _ = Describe("With a running MachineSetSync controller", func() {
 							))),
 					)
 				})
+
+				Context("when the MAPI machine set has a non-zero deletion timestamp", func() {
+					BeforeEach(func() {
+						Eventually(k.Object(mapiMachineSet), timeout).Should(
+							HaveField("ObjectMeta.Finalizers", ContainElement(machinesync.SyncFinalizer)),
+						)
+						Eventually(k.Object(capiMachineSet), timeout).Should(
+							SatisfyAll(
+								HaveField("ObjectMeta.Finalizers", ContainElement(machinesync.SyncFinalizer)),
+								HaveField("ObjectMeta.Finalizers", ContainElement(capiv1beta1.MachineSetFinalizer)),
+							),
+						)
+						Expect(k8sClient.Delete(ctx, mapiMachineSet)).To(Succeed())
+					})
+					// Expect to see the finalizers, so they're in place before
+					//  we Expect logic that relies on them to work
+					It("should delete the CAPI machine set", func() {
+						Eventually(k.Get(capiMachineSet), timeout).Should(WithTransform(apierrors.IsNotFound, BeTrue()))
```

Comment:
On anything that uses `BeTrue()` it's doubly important to add context, so that when the test fails, we don't just get `expected true to be false`
```suggestion
						Eventually(k.Get(capiMachineSet), timeout).Should(WithTransform(apierrors.IsNotFound, BeTrue()), "eventually capiMachineSet should not be found")
```

Can you do something like ^ on all the similar cases you've got here?